test/canvas-mock.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Set up mock 2D context, for usage in unit tests.
+ *
+ * Adapted from: https://github.com/Cristy94/canvas-mock
+ */
+
+var myHackedInCanvas;    // Use one global canvas instance for all calls to createElement('canvas');
+
+
+function replaceCanvasContext (el) {
+	el.getContext = function() {
+		return {
+			fillRect: function() {},
+			clearRect: function(){},
+			getImageData: function(x, y, w, h) {
+				return  {
+					data: new Array(w*h*4)
+				};
+			},
+			putImageData: function() {},
+			createImageData: function(){ return []},
+			setTransform: function(){},
+			drawImage: function(){},
+			save: function(){},
+			fillText: function(){},
+			restore: function(){},
+			beginPath: function(){},
+			moveTo: function(){},
+			lineTo: function(){},
+			closePath: function(){},
+			stroke: function(){},
+			translate: function(){},
+			scale: function(){},
+			rotate: function(){},
+			arc: function(){},
+			fill: function(){},
+
+      //
+      // Following added for vis.js unit tests
+      //
+
+      measureText: function(text) {
```

Comment:
Spacing looks off

Reply:
FYI, this code is a module which I took from [here](https://github.com/Cristy94/canvas-mock) and adapted it to own needs when I realized it wasn't completely fit for the purpose.

If think you mean this spacing?

```js
      measureText: function(text) {
        return {
          width:>>  <<12*text.length,
          height: 14
        };
      },
};

```

Otherwise, I don't know what you mean. I have a habit of aligning related stuff, but indeed it's not needed here with only two lines.

----
Ah wait, there's tabs there. That's probably what you mean. Ah, right, that's from the original. I'll change it to spaces.